### PR TITLE
Reformat statistical: mean function added (Issue #1890)

### PR DESCRIPTION
**Pull request by sakibkuet10**
Issue #1890 

reformatted mean function: added docstring and examples for mean function in ivy, array and container

### CODE REVIEW

File: ivy/array/statistical.py [hunk 1]
```diff
@@ -33,10 +33,84 @@ def mean(
...
+            must have the same data type as ``x``.
+
+        Examples
+        --------
+        >>> x = ivy.array([3., 4., 5.])
+        >>> y = x.mean()
+        >>> print(y)
+        ivy.array(4.)
+
+        >>> x = ivy.array([3., 4., 5.])
```
Comment (jiahanxie353) [inline ivy/array/statistical.py:83]: Could you vary the inputs a little bit? Repeating inputs are not helpful for others who will be reading your docstrings.
Also, pls add one sentence comment above each example to denote which kind of example it belongs to. For instance: `With :code:`ivy.Array` input:`. you can refer to the docstring section of [Deep Dive](https://lets-unify.ai/ivy/deep_dive/13_docstring_examples.html)

File: ivy/container/statistical.py [hunk 2]
```diff
@@ -64,14 +64,111 @@ def mean(
...
+        -------
+        ret
+           container, if the arithmetic mean was computed over the entire array,
+           a zero-dimensional array containing the arithmetic mean;
+           otherwise, a non-zero-dimensional array containing the arithmetic
+           means. The returned array must have the same data type as ``self``.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3., 4., 5.]))
```
Comment (jiahanxie353) [inline ivy/container/statistical.py:120]: Same comment here,

1. avoid repeating inputs;
2. add a one sentence description above

File: ivy/functional/ivy/statistical.py [hunk 3]
```diff
@@ -174,6 +174,219 @@ def mean(
            array ``x`` has an integer data type, the returned array must have the
            default floating-point data type.
 
+    Examples
+    --------
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([3., 4., 5.])
```
Comment (jiahanxie353) [inline ivy/functional/ivy/statistical.py:181]: Same comment here,

1. avoid repeating inputs;
2. add a one sentence description above

**Review by jiahanxie353 — CHANGES_REQUESTED** [review]
Left some comments below. Please revise accordingly. Thanks for your contribution

File: ivy/array/statistical.py (hunk 1 above)
Comment (sakibkuet10) [reply ivy/array/statistical.py:83]: Thanks @jiahanxie353  for your feedback. I will update accordingly

File: ivy/functional/ivy/statistical.py (hunk 3 above)
Comment (sakibkuet10) [reply ivy/functional/ivy/statistical.py:181]: Hi, here each example is denoted which kind of example it belongs to. I am just modifying the repeating inputs.

**Comment by sakibkuet10** [comment]
> Left some comments below. Please revise accordingly. Thanks for your contribution

Let me know if there is anything need to be changed. Thanks for your review & feedback

**Comment by sakibkuet10** [comment]
> > Left some comments below. Please revise accordingly. Thanks for your contribution
> 
> I have updated the files according to your feedbacks. Thanks @jiahanxie353 for your review & feedback

Hi @jiahanxie353 should I change anything?

**Comment by roboFiddle** [comment]
This looks good, merging. Thanks for the contribution!

**Comment by sakibkuet10** [comment]
> This looks good, merging. Thanks for the contribution!

Thanks

**Pull request merged**